### DISCUSSION TITLE
How to implement global exception handler in Kotlin?

### DISCUSSION THREAD

**Discussion by jehrenzweig-leagueapps** [micronaut-projects/micronaut-core#7496] [Q&A]
I want to return a custom exception object when any of my Micronaut implementation methods throws an unhandled exception.  I've seen [the guide on how to do it using Java](https://guides.micronaut.io/latest/micronaut-error-handling-gradle-java.html), but I'm stumbling when I try to implement this approach using Kotlin.

Here's what my code currently looks like:
```
import com.leagueapps.microservices.poc.potato.models.CustomException
import io.micronaut.context.annotation.Requires;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.HttpStatus
import io.micronaut.http.annotation.Produces;
import io.micronaut.http.server.exceptions.ExceptionHandler;
import jakarta.inject.Singleton;

@Produces
@Singleton
@Requires(classes = [ RuntimeException::class.java ])
public class GlobalExceptionHandler : ExceptionHandler<RuntimeException, HttpResponse<Any>> {
	override fun handle(request: HttpRequest<*>?, exception: RuntimeException?): HttpResponse<Any> {
                val customException = CustomException(summary = "OMG BADNESS HAPPENED", detail = exception?.message)
		return HttpResponse.status<CustomException>(HttpStatus.INTERNAL_SERVER_ERROR).body(customException);
	}
}
```

The `GlobalExceptionHandler` class that I've declared never gets called right now, if/when an API controller method throws an exception.  Am I missing a step (or few) somewhere?

**Comment by morki** [comment]
The annotation part of your problem is solved by this syntax:

```kotlin
@Requires(classes = [RuntimeException::class])
```

**Reply by jehrenzweig-leagueapps** [reply]
> The annotation part of your problem is solved by this syntax:
> 
> ```kotlin
> @Requires(classes = [RuntimeException::class])
> ```

Thank you!  IntelliJ did nothing to point out that my array declaration was using incorrect syntax -- it just told me the annotation was bad, which was a red herring.

I've updated my original post to reflect the fix, to help make the question focused primarily on the exception handler not firing.
